### PR TITLE
Use libipld instead of serde everywhere

### PR DESCRIPTION
**Pull request by rklaehn**

### CODE REVIEW

File: banyan-cli/src/tag_index.rs [hunk 1]
```diff
@@ -23,6 +28,29 @@ pub struct TagIndex {
     pub(crate) elements: Vec<IndexSet>,
 }
 
+impl Encode<DagCborCodec> for TagIndex {
+    fn encode<W: std::io::Write>(&self, c: DagCborCodec, w: &mut W) -> anyhow::Result<()> {
+        todo!()
```
Comment (dvc94ch) [inline banyan-cli/src/tag_index.rs:33]: ready for review? :)

Comment (rklaehn) [reply banyan-cli/src/tag_index.rs:33]: oh well, you are right. I was thinking about getting rid of the whole banyan-cli thing and instead having an ax trees cli inside cosmos.

Comment (rklaehn) [reply banyan-cli/src/tag_index.rs]: Implemented it in a very inefficient way for now. I might add libipld support to my vec_collections in the longer term.

File: banyan/Cargo.toml [hunk 2]
```diff
@@ -36,3 +34,4 @@ quickcheck_macros = "1.0.0"
 sha2 = "0.9.3"
 tokio = { version = "1.2.0", features = ["full"] }
 tracing-subscriber = "0.2.15"
+rand_chacha = "0.3.0"
```
Comment (dvc94ch) [inline banyan/Cargo.toml:37]: now that you've updated cargo-edit you can remove it and watch the magic of cargo-add at work

**Review by dvc94ch — APPROVED** [review]
looks good

File: banyan/Cargo.toml (hunk 2 above)
Comment (rklaehn) [reply banyan/Cargo.toml]: Amazing technology...

**Pull request merged**